### PR TITLE
[#157245263] Initial Push Notification integration

### PR DESCRIPTION
**Pull request by fpersico**
Add initial PushNotification integration on iOS and Android using the [react-native-push-notification](https://github.com/zo0r/react-native-push-notification) library.

### CODE REVIEW

File: ts/boot/configurePushNotification.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      if (debugRemotePushNotification) {
+        Alert.alert("Notification", JSON.stringify(notification));
+      }
+
+      // On iOS we need to call this when the remote notification handling is complete
+      notification.finish(PushNotificationIOS.FetchResult.NoData);
+    },
+
+    // GCM Sender ID
+    senderID: "317157111831"
```
Comment (gunzip) [inline ts/boot/configurePushNotification.ts]: why not from env ?

File: .env.production [hunk 2]
```diff
@@ -4,4 +4,5 @@ ENVIRONMENT=PROD
 API_URL_PREFIX='https://app-backend.k8s.test.cd.teamdigitale.it'
 MIXPANEL_TOKEN='b5c28d653684157f831d6b604d5e1956'
 ENABLE_TEST_IDP=NO
+GCM_SENDER_ID='317157111831'
```
Comment (cloudify) [inline .env.production]: I think we will use a different one for production - @gunzip ?

File: ts/App.tsx [hunk 3]
```diff
@@ -7,13 +7,15 @@ import { PersistGate } from "redux-persist/integration/react";
...
 import configureErrorHandler from "./boot/configureErrorHandler";
+import configurePushNotifications from "./boot/configurePushNotification";
 import configureStoreAndPersistor from "./boot/configureStoreAndPersistor";
 import * as config from "./config";
 import RootContainer from "./RootContainer";
 
 // Inizialize Mixpanel and configure the global js error handler
 Mixpanel.sharedInstanceWithToken(config.mixpanelToken);
 configureErrorHandler();
+configurePushNotifications();
```
Comment (cloudify) [inline ts/App.tsx]: btw not sure what is the best practice here, but there's something we need to fox also for `configureErrorHandler`: both return a `Promise` and `Promise`s can't be awaited in the global scope (see [this tslint rule](https://palantir.github.io/tslint/rules/no-floating-promises/))

Comment (cloudify) [reply ts/App.tsx]: actually I just noticed that `configurePushNotifications` doesn't return a `Promise`, my point stands for `configureErrorHandler`

File: .env.production (hunk 2 above)
Comment (gunzip) [reply .env.production]: AFAIK Google does not require different sender for test / production and Apple requires different keys (not sender id) (but I'm not an expert)

**Pull request merged**